_posts/2019-08-13-Using-Lets-Encrypt-Certificates-In-Java.md

```diff
@@ -0,0 +1,408 @@
+---
+layout: post
+authors: [ken_coenen]
+title: 'Using Let''s Encrypt certificates in Java applications'
+image: /img/lets-encrypt.png
+tags: [Security, TLS, Let's Encrypt, Keycloak]
+category: Security
+comments: true
+---
+
+> At some point in their career, developers come accross the need to work with security certificates.
+This article describes how to setup Let's Encrypt, retrieve a certificate, renew it automatically and how to use the certificate in a Java application for TLS communication.
+
+# Table of contents
+1. [Certificate Authorities and Let's Encrypt](#certificate-authorities-and-lets-encrypt)
+2. [Installing a Let's Encrypt certificate](#installing-a-lets-encrypt-certificate)
+3. [Certificate renewal](#certificate-renewal)
+4. [Automating the renewal process](#automating-the-renewal-process)
+5. [Using keystores and truststores in a Java application](#using-keystores-and-truststores-in-a-java-application)
+6. [Resources](#resources)
+
+# Certificate Authorities and Let's Encrypt
+
+When you want to enable HTTPS on your website or need certificates for TLS communication, you'll need to request this certificate from a [Certificate Authority](https://en.wikipedia.org/wiki/Certificate_authority){:target="_blank" rel="noopener noreferrer"} (CA).
+It acts as a [trusted third party](https://en.wikipedia.org/wiki/Trusted_third_party){:target="_blank" rel="noopener noreferrer"} between two parties that need to communicate with each other.
+[Let's Encrypt](https://certbot.eff.org/){:target="_blank" rel="noopener noreferrer"} is such a Certificate Authority.
+It is their mission to give everyone a secure and privacy-respecting web experience.
+That's why they issue certificates free of charge.
+
+# Installing a Let's Encrypt certificate
+
+We assume that you have shell access to your server.
```

Comment:
Suggestion: Assuming that you have shell access to your server, Let's Encrypt recommends ...